WordPress/Classes/Utility/ContextManager+ErrorHandling.swift

```diff
@@ -0,0 +1,109 @@
+    NSPersistentStoreSaveError: "Unclassified save error - something we depend on returned an error",
+    NSPersistentStoreTimeoutError: "Failed to connect to the persistent store within the specified timeout (see NSPersistentStoreTimeoutOption)",
+    NSPersistentStoreTypeMismatchError: "Returned by persistent store coordinator if a store is accessed that does not match the specified type",
+    NSPersistentStoreUnsupportedRequestTypeError: "An NSPersistentStore subclass was passed an NSPersistentStoreRequest that it did not understand",
+    NSSQLiteError: "General SQLite error ",
+    NSValidationDateTooLateError: "Some date value is too late",
+    NSValidationDateTooSoonError: "Some date value is too soon",
+    NSValidationInvalidDateError: "Some date value fails to match date pattern",
+    NSValidationInvalidURIError: "Some URI value cannot be represented as a string",
+    NSValidationMissingMandatoryPropertyError: "Non-optional property with a nil value",
+    NSValidationMultipleErrorsError: "Generic message for error containing multiple validation errors",
+    NSValidationNumberTooLargeError: "Some numerical value is too large",
+    NSValidationNumberTooSmallError: "Some numerical value is too small",
+    NSValidationRelationshipDeniedDeleteError: "Some relationship with NSDeleteRuleDeny is non-empty",
+    NSValidationRelationshipExceedsMaximumCountError: "Bounded, to-many relationship with too many destination objects",
+    NSValidationRelationshipLacksMinimumCountError: "To-many relationship with too few destination objects",
+    NSValidationStringPatternMatchingError: "Some string value fails to match some pattern",
+    NSValidationStringTooLongError: "Some string value is too long",
+    NSValidationStringTooShortError: "Some string value is too short",
+]
+
+private extension NSExceptionName {
+    static let coreDataSaveMainException = NSExceptionName("Unresolved Core Data save error (Main Context)")
+    static let coreDataSaveDerivedException = NSExceptionName("Unresolved Core Data save error (Derived Context)")
+}
+
+extension ContextManager {
+    @objc(handleSaveError:inContext:)
+    func handleSaveError(_ error: NSError, in context: NSManagedObjectContext) {
+        let isMainContext = context == mainContext
+        let exceptionName: NSExceptionName = isMainContext ? .coreDataSaveMainException : .coreDataSaveDerivedException
+        let reason = reasonForError(error)
+        DDLogError("Unresolved Core Data save error: \(error)")
+        DDLogError("Generating exception with reason:\n\(reason)")
+        // Sentry is choking when userInfo is too big and not sending crash reports
+        // For debugging we can still see the userInfo details since we're logging the full error above
+        let exception = NSException(name: exceptionName, reason: reason, userInfo: nil)
```

Comment:
The thing is, even if it fits and gets sent, I haven't been able to find anything from `userInfo` in the Sentry UI 🤷‍♂ 

And in any case, it would contain the full object that failed to validate, which in turn could potentially have some sensitive/private information. I'd rather keep that in the logs that only get sent to us when users contact support.